src/hooks/use-snapshots.tsx

```diff
@@ -280,13 +284,16 @@ export const SnapshotProvider: React.FC< { children: ReactNode } > = ( { childre
 		}
 		setIsLoading( true );
 		try {
-			const response: { site_count: number; site_limit: number } = await client.req.get( {
-				path: '/jurassic-ninja/usage',
-				apiNamespace: 'wpcom/v2',
-			} );
+			const response: { site_count: number; site_limit: number; site_creation_blocked: boolean } =
+				await client.req.get( {
+					path: '/jurassic-ninja/usage',
+					apiNamespace: 'wpcom/v2',
+				} );
+			setSnapshotCreationBlocked( response.site_creation_blocked );
 			return response;
 		} catch ( error ) {
 			Sentry.captureException( error );
+			setSnapshotCreationBlocked( false ); // Default the site creation to enabled
```

Comment:
Not sure if in case of an error when getting the data from the endpoint, we should keep the creation disabled or enabled. My thinking was to set it to enabled regardless so that the user could attempt the action and in the worst case scenario, they would get a dialog window with an error.

Happy to change it though if there is a better approach

Reply:
Probably we don't need to change the value when the request fails, as we don't know the value to use. Ultimately, if the user is blocked, the site won't be created on the server's end.

Reply:
Sounds good, I made chanegs in https://github.com/Automattic/studio/pull/354/commits/bba74b57a5120e21952eba668971e210eb2e3ee5